### PR TITLE
Cherry-pick "[M68k] Fix ODR violation in GISel code (#72797)"

### PR DESCRIPTION
**Pull request by petrochenkov**
https://github.com/llvm/llvm-project/commit/8cb842879d44ba7b5c1d3e158b8824cae1db6092

To get `download-ci-llvm` to work sooner (https://github.com/rust-lang/rust/issues/107668).

### CODE REVIEW

**Comment by petrochenkov** [comment]
Not sure what is the expected process for such cherry-picking and updating LLVM in rust-lang/rust.

**Comment by cuviper** [comment]
You're within the correct process:
https://rustc-dev-guide.rust-lang.org/backend/updating-llvm.html#bugfix-updates

Next will be a PR with the submodule update in the rust repo.

File: llvm/lib/Target/M68k/GISel/M68kCallLowering.cpp [hunk 1]
```diff
@@ -25,6 +25,27 @@
...
+};
+
+struct CallReturnHandler : public M68kIncomingValueHandler {
+  CallReturnHandler(MachineIRBuilder &MIRBuilder, MachineRegisterInfo &MRI,
+                    MachineInstrBuilder &MIB)
+      : M68kIncomingValueHandler(MIRBuilder, MRI), MIB(MIB) {}
+
+private:
+  void assignValueToReg(Register ValVReg, Register PhysReg,
+                        const CCValAssign &VA) override;
```
Comment (nikic) [inline llvm/lib/Target/M68k/GISel/M68kCallLowering.cpp:42]: This needs to be `CCValAssign VA` for LLVM 17. This causes the build failure in https://github.com/rust-lang/rust/pull/119159.

Comment (petrochenkov) [reply llvm/lib/Target/M68k/GISel/M68kCallLowering.cpp:42]: Sigh.
Fixed in https://github.com/rust-lang/llvm-project/pull/160.

**Pull request merged**